MergeAndCheckCategoryTest

sarubhai opened the pull request:
Add if you feel anything more required. Else I will follow the same for other models

sarubhai committed:
MediaController

sarubhai committed:
CheckCategoryTesting

sarubhai committed:
Merge branch 'master' of https://github.com/sarubhai/platonics2

akash-mitra commented on tests/Unit/CategoriesTest.php:
```diff
@@ -7,41 +7,136 @@
     }
 
-    public function test_that_store_method_can_persist_data()
+    public function test_update_can_persist_data()
```
This is not the correct approach. You need to update a already created category during `TestDataSetUp`

akash-mitra commented on tests/Unit/CategoriesTest.php:
```diff
@@ -7,41 +7,136 @@
+            'description' => 'test category desc',
+            'access_level' => 'F'
+        ];
+        $response = $this->post('/api/categories', $category)->decodeResponseJson();
```
Here you are creating a category first for the purpose of updating later. Problem is - If there is any issue in `create` method but no issue in `update` method, then also update test case will fail. Since these are unit tests, you can not put two different functionalities together.

akash-mitra commented on app/Http/Controllers/MediaController.php:
```diff
@@ -4,7 +4,101 @@
+    {
+        $media = Media::FindOrFail($id);
+        $media->delete();
+        return response()->json(null, 204); 
```
Not returning media `name`?

akash-mitra reviewed (changes requested):
Review my comments.

sarubhai committed:
TestCasesDone

Memory Testing

Remove local testing.sqlite

php artisan config:clear
php artisan cache:clear
./vendor/bin/phpunit tests

Also change actual DB:

php artisan migrate:rollback
php artisan migrate:refresh --seed
